chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -374,171 +390,148 @@ where
 
     fn call(
         &self,
-        logger: &Logger,
+        logger: Logger,
         contract_address: Address,
         call_data: Bytes,
-        block_number_opt: Option<BlockNumber>,
+        block_ptr: EthereumBlockPointer,
     ) -> impl Future<Item = Bytes, Error = EthereumContractCallError> + Send {
-        let block_number_opt = block_number_opt.map(Into::into);
         let web3 = self.web3.clone();
-        let logger = logger.clone();
 
-        // Outer retry used only for 0-byte responses,
-        // where we can't guarantee the problem is temporary.
-        // If we keep getting back 0-byte responses,
-        // eventually we assume it's right and return it.
-        retry("eth_call RPC call (outer)", &logger)
-            .when(|result: &Result<Bytes, _>| {
-                match result {
-                    // Retry only if zero-length response received
-                    Ok(bytes) => bytes.0.is_empty(),
-
-                    // Errors are retried in the inner retry
-                    Err(_) => false,
-                }
+        // Ganache does not support calls by block hash.
+        // See https://github.com/trufflesuite/ganache-cli/issues/745
+        let block_id = if self.is_ganache {
+            BlockId::Number(block_ptr.number.into())
+        } else {
+            BlockId::Hash(block_ptr.hash)
+        };
+
+        retry("eth_call RPC call", &logger)
+            .when(|result| match result {
+                Ok(_) | Err(EthereumContractCallError::Revert(_)) => false,
+                Err(_) => true,
             })
-            .limit(16)
-            .no_logging()
-            .no_timeout()
+            .limit(10)
```

Comment:
What impact does this have in situations where the Ethereum node is temporarily down? Do we still keep retrying indefinitely on e.g. network errors, or do subgraphs fail eventually, after 10 attempts?